Engineering/UnitTesting.md

```diff
@@ -1,41 +1,56 @@
 
-Bug fixes should start with a test that reliably reproduces the bug to ensure that a particular commit will fix the bug as intended. Existing tests will reduce risk of regressions introduced by the fix.
+Bug fixes should start with a test that reliably reproduces the bug. This will help ensure that the code correctly fixes the issue and that it is not reintroduced in future changes. Existing tests will reduce risk of regressions introduced by the fix.
 
-Unit testing works in conjunction with integration tests and end-to-end tests for larger pieces of functionality.
+Developers should be careful to focus their unit testing efforts on the code that they are currently working on. Writing tests to validate that the functionality of a 3rd party framework should be considered out of scope. An integration test to ensure handling 3rd party failures is acceptable, but should be focused on how internal code handles the failure, not on how the external failure triggered. It is easy to bogged down in writing unit tests that test functionality that is not in the scope of responsibility of the developer. For example, writing a test to validate the implementation of an framework's function for parsing a number from a string value.
 
-In order to keep execution of unit tests fast and executable as part of a CI/CD pipeline, tests can provide mock implementations of other parts of the application or 3rd party services. 
+In order to keep execution of unit tests fast and executable as part of a CI/CD pipeline, tests should provide mock implementations of dependencies including other parts of the application or 3rd party services. 
 
-For integration or end-to-end testing, mocks should be replaced with API calls to the system they are simulating.
+Unit testing represents only the foundational testing in the system development. It does not remove the need for integration tests and end-to-end tests for larger pieces of functionality and system dependencies. For integration or end-to-end testing, mocks should be replaced with API calls to the system they are simulating.
 
 ### Writing Tests
 Good unit tests follow a few general principles:
-- Pass / Fail tests ensure intended succeeds and fails as designed.
-- Transactional tests ensure transactions commit or roll back as designed
-- CRUD operations work as designed
+- Run incredibly fast
+- Test only the current system under test
+- Test one thing generally with a single assertion
+- Have only one reason to fail
+
+Consider an integration test:
+- Ensure transactions commit or roll back as designed
+- Ensure CRUD operations work as designed
 - All data created for or during a test is localized to the test to allow for parallel test execution
 - All data created during or for a test gets cleaned up after tests completed
 
-For more complex applications, unit tests also ensure:
-- that multi-threading works as intended,
+For more complex applications, integration load tests can also ensure:
```

Comment:
if the scope of the test is bound to the unit, I would consider this still to be unit test (and not integration tests)